src/components/chat/chat_messages.jsx

```diff
@@ -27,21 +25,23 @@ const ChatMessages = React.createClass({
     }
   },
 
-  componentWillReceiveProps() {
+  componentWillReceiveProps(p) {
     let previousChannel = this.state.channel;
+
     this.setState({
       channel: this.getParams().roomSlug
     }, () => {
-      if (previousChannel !== this.state.channel) {
-        ChatActions.destroy();
-        ChatActions.init(
-          `${AppStore.getUrl()}/rooms/${this.state.channel}/messages`,
-          CurrentUserStore.getToken()
-        );
-      }
+      this.updateMessages();
+      this.handleChannelChange()
     });
   },
 
+  handleChannelChange(){
```

Comment:
hmmm, it might... Than I'll have to find a different trigger. I'm already automatically fetching the message history for all the channels the user is subscribed to,which catches a lot of these cases. The only thing we'd still have to catch is a user entering a room they're not subscribed to. (can they?)